File: src/main/java/com/thebluehats/server/core/PerksService.java
package com.thebluehats.server.core;

import com.google.inject.Injector;
import com.thebluehats.server.game.managers.world.PerkManager;
import com.thebluehats.server.game.perks.Perk;
import com.thebluehats.server.game.perks.Vampire;
import com.thebluehats.server.game.utils.Registerer;

public class PerksService implements Service {
    @Override
    public void provision(Injector injector) {
        Registerer<Perk> registerer = injector.getInstance(PerkManager.class);

        registerer.register(new Perk[] { injector.getInstance(Vampire.class) });
    }
}

File: minecraft-server/src/main/kotlin/com/thebluehats/server/core/services/PerksService.kt
package com.thebluehats.server.core.services

import com.google.inject.Injector
import com.thebluehats.server.game.managers.world.PerkManager
import com.thebluehats.server.game.perks.Perk
import com.thebluehats.server.game.perks.Vampire
import com.thebluehats.server.game.utils.Registerer

class PerksService : Service {
    override fun provision(injector: Injector) {
        val registerer: Registerer<Perk> = injector.getInstance(PerkManager::class.java)
        registerer.register(injector.getInstance(Vampire::class.java))
    }
}